### PR TITLE
Adding Metadata Service Manager

### PR DESCRIPTION
**Pull request by osaidwtd**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: metadata/services/graphiql.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func (m *metadataService) Query(query QueryDetails) ([]byte, error) {
+	graphiqlUrl, err := url.Parse(m.GetMetaDataDetails().GetUrl() + "api/v1/query")
+	if err != nil {
+		return []byte{}, errorutils.CheckError(err)
+	}
+
+	httpClientDetails := m.GetMetaDataDetails().CreateHttpClientDetails()
+	rtUtils.SetContentType("application/json", &httpClientDetails.Headers)
```
Comment (sverdlov93) [inline metadata/services/graphiql.go:36]:
```suggestion
	httpClientDetails.SetContentTypeApplicationJson()
```

File: metadata/services/graphiql.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+func NewMetadataService(serviceDetails auth.ServiceDetails, client *jfroghttpclient.JfrogHttpClient) Service {
+	return &metadataService{serviceDetails: &serviceDetails, client: client}
+}
+
+func (m *metadataService) GetMetaDataDetails() auth.ServiceDetails {
+	return *m.serviceDetails
+}
+
+func (m *metadataService) Query(query QueryDetails) ([]byte, error) {
+	graphiqlUrl, err := url.Parse(m.GetMetaDataDetails().GetUrl() + "api/v1/query")
```
Comment (sverdlov93) [inline metadata/services/graphiql.go:30]:
```suggestion
	graphiqlUrl, err := url.Parse(m.GetMetadataDetails().GetUrl() + "api/v1/query")
```

File: metadata/services/graphiql.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	if err != nil {
+		return []byte{}, errorutils.CheckError(err)
+	}
+
+	httpClientDetails := m.GetMetaDataDetails().CreateHttpClientDetails()
```
Comment (sverdlov93) [inline metadata/services/graphiql.go:35]:
```suggestion
	httpClientDetails := m.GetMetadataDetails().CreateHttpClientDetails()
```

File: tests/timeout_test.go [hunk 4]
```diff
@@ -136,6 +141,27 @@ func testEvidenceTimeout(t *testing.T) {
...
+	servicesManager, err := metadata.NewManager(createServiceConfigWithTimeout(t, details))
+	assert.NoError(t, err)
+
+	query := []byte("query body")
+	serviceDetails := metadataServices.QueryDetails{
+		Body: query,
+	}
+
+	// Expect timeout
+	_, err = servicesManager.GraphiqlQuery(serviceDetails)
```
Comment (sverdlov93) [inline tests/timeout_test.go:161]: GraphiqlQuery ? not GraphqlQuery?
same for the name of the file graphiql.go

File: metadata/services/graphiql.go (hunk 2 above)
Comment (sverdlov93) [inline metadata/services/graphiql.go:30]: lets use `api/v1/query` as const var
also why `url.parse` and not just `path.join`?

File: metadata/services/graphiql_test.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	client, err := jfroghttpclient.JfrogClientBuilder().Build()
+	assert.NoError(t, err)
+	return testServer, &metadataService{serviceDetails: &serviceDetails, client: client}
+}
+
+func createMetadataHandlerFunc(t *testing.T) (http.HandlerFunc, *int) {
+	requestNum := 0
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/api/v1/query" {
```
Comment (sverdlov93) [inline metadata/services/graphiql_test.go:42]: lets get the url from a const var

File: README.md [hunk 6]
```diff
@@ -244,6 +244,13 @@
       - [Creating New Evidence Service Manager](#creating-new-evidence-service-manager)
     - [Using Evidence Services](#using-evidence-services)
       - [Upload Evidence](#upload-evidence)
+  - [Evidence APIs](#evidence-apis)
+    - [Creating Evidence Service Manager](#creating-evidence-service-manager)
+      - [Creating Evidence Details](#creating-evidence-details)
+      - [Creating Evidence Service Config](#creating-evidence-service-config)
+      - [Creating New Evidence Service Manager](#creating-new-evidence-service-manager)
+    - [Using Evidence Services](#using-evidence-services)
+      - [Upload Evidence](#upload-evidence)
```
Comment (sverdlov93) [inline README.md:253]: ?

Comment (osaidwtd) [reply README.md:253]: done

File: metadata/services/graphiql_test.go (hunk 5 above)
Comment (osaidwtd) [reply metadata/services/graphiql_test.go:42]: done

File: metadata/services/graphiql.go (hunk 2 above)
Comment (osaidwtd) [reply metadata/services/graphiql.go:30]: done

File: tests/timeout_test.go (hunk 4 above)
Comment (osaidwtd) [reply tests/timeout_test.go:161]: done

File: metadata/services/graphiql.go (hunk 3 above)
Comment (osaidwtd) [reply metadata/services/graphiql.go:35]: done

File: metadata/services/graphiql.go (hunk 2 above)
Comment (osaidwtd) [reply metadata/services/graphiql.go:30]: done

File: metadata/services/graphiql.go (hunk 1 above)
Comment (osaidwtd) [reply metadata/services/graphiql.go:36]: can't find it

File: metadata/manager.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+	return manager, err
+}
+
+func (mm *metadataManager) Client() *jfroghttpclient.JfrogHttpClient {
+	return mm.client
+}
+
+func (mm *metadataManager) GraphqlQuery(query []byte) ([]byte, error) {
+	evidenceService := services.NewMetadataService(mm.config.GetServiceDetails(), mm.client)
```
Comment (sverdlov93) [inline metadata/manager.go:43]:
```suggestion
	metadataService := services.NewMetadataService(mm.config.GetServiceDetails(), mm.client)
```

Comment (osaidwtd) [reply metadata/manager.go]: done

**Pull request merged**